plugins/chat/lib/chat/transcript_service.rb

```diff
@@ -166,9 +166,17 @@ def generate_markdown
       previous_message = nil
       rendered_markdown = []
       rendered_thread_markdown = []
-      all_messages_same_user = messages.count(:user_id) == 1
       threading_enabled = @channel.threading_enabled?
       thread_id = threading_enabled ? messages.first.thread_id : nil
+      thread = Chat::Thread.find_by(id: thread_id) if thread_id.present? && threading_enabled
+
+      # We are getting only the OP of the thread, let's expand it to
+      # include all the replies for the thread.
+      if messages.count == 1 && thread&.original_message_id == messages.first.id
+        @messages = [messages.first] + messages.first.thread.replies
+      end
+
+      all_messages_same_user = messages.count(:user_id) == 1
```

Comment:
I think because we might have replaced the relation in `@messages` with an array, this might not do what we think. `Array#count` will look for the exact argument, i.e. counting the number of `:user_id` symbols in the array.

Reply:
Ah thanks for picking that up, will add a fix + test